### PR TITLE
Facet docs

### PR DESCRIPTION
**Pull request by coszio**
[Rendered](https://deploy-preview-1178--condescending-goldwasser-91acf0.netlify.app/documentation/concepts/payload/#facet-counts)

Documents the new Facet API, to be introduced in v1.12.0

### TODO
- [x] Add snippets for
  - [x] Rust
  - [x] Typescript
  - [x] CSharp
  - [x] Java
  - [x] Go

### CODE REVIEW

File: qdrant-landing/content/documentation/concepts/payload.md [hunk 1]
```diff
@@ -1273,3 +1273,83 @@ Payload schema example:
...
+      {"value": "XL", "count": 1},
+      {"value": "XXL", "count": 1}
+    ]
+  },
+  "time": 0.0001
+}
+```
+The results are sorted by the count in descending order, then by the value in ascending order.
+
+As a security measure, only values with non-zero counts will be returned.
```
Comment (agourlay) [inline qdrant-landing/content/documentation/concepts/payload.md:1335]: Also does faceting require a payload index?

File: qdrant-landing/content/documentation/concepts/payload.md [hunk 2]
```diff
@@ -1273,3 +1273,83 @@ Payload schema example:
     }
 }
 ```
+
+## Facet counts
+
+Faceting is a special counting technique that can be used for various purposes:
+- Know which unique values exist in a field.
```
Comment (timvisee) [inline qdrant-landing/content/documentation/concepts/payload.md:1280]: Obviously nitpicking, but I believe we use this working in other places:

```suggestion
- Know which unique values exist for a payload key.
```

File: qdrant-landing/content/documentation/concepts/payload.md (hunk 1 above)
Comment (timvisee) [reply qdrant-landing/content/documentation/concepts/payload.md:1335]: Yes, it is noted above:

```markdown
<aside role="status">In Qdrant you can facet on a field <strong>only</strong> if you have created a field index that supports <code>MatchValue</code> conditions for it, like a <code>keyword</code> index.</aside>
```

Comment (agourlay) [reply qdrant-landing/content/documentation/concepts/payload.md:1335]: Thanks, my brain skipped the markup line completely :face_with_head_bandage:

File: qdrant-landing/content/documentation/concepts/payload.md [hunk 3]
```diff
@@ -1273,3 +1273,83 @@ Payload schema example:
...
+      }
+    }
+}
+```
+
+```python
+client.facet(
+    collection_name="{collection_name}",
+    key="size",
+    filter=Filter(must=[Match("color", "red")]),
```
Comment (agourlay) [inline qdrant-landing/content/documentation/concepts/payload.md:1311]: the field is called `facet_filter` :)

File: qdrant-landing/content/documentation/concepts/payload.md [hunk 4]
```diff
@@ -1273,3 +1273,83 @@ Payload schema example:
...
+)
+```
+
+// TODO: add the rest of the client examples
```
Comment (agourlay) [inline qdrant-landing/content/documentation/concepts/payload.md:1315]: Java example

```java
 client
      .facetAsync(
          Points.FacetCounts.newBuilder()
              .setCollectionName(collection_name)
              .setFilter(Filter.newBuilder().addMust(matchKeyword("color", "red")).build())
              .setKey("foo")
              .build())
      .get();
  ```

**Review by generall — APPROVED** [review]

**Pull request merged**